public/app/AppWrapper.tsx

```diff
@@ -144,3 +142,109 @@ export class AppWrapper extends Component<AppWrapperProps, AppWrapperState> {
     );
   }
 }
+
+function RouterTree(props: { routes?: JSX.Element | false }) {
+  return (
+    <Router history={locationService.getHistory()}>
+      <LocationServiceProvider service={locationService}>
+        <CompatRouter>
+          <ModalsContextProvider>
+            <GlobalStyles />
+            <AppChrome>
+              <AngularRoot />
+              <AppNotificationList />
+              <Stack gap={0} grow={1} direction="column">
+                {pageBanners.map((Banner, index) => (
+                  <Banner key={index.toString()} />
+                ))}
+                {props.routes}
+              </Stack>
+              {bodyRenderHooks.map((Hook, index) => (
+                <Hook key={index.toString()} />
+              ))}
+            </AppChrome>
+            <LiveConnectionWarning />
+            <ModalRoot />
+            <PortalContainer />
+          </ModalsContextProvider>
+        </CompatRouter>
+      </LocationServiceProvider>
+    </Router>
+  );
+}
+
+/**
+ * Renders both the main app tree and a secondary sidecar app tree to show 2 apps at the same time in a resizable split
+ * view.
+ * @param props
+ * @constructor
+ */
+function ExperimentalSplitPaneTree(props: { routes?: JSX.Element | false }) {
+  const { activePluginId, closeApp } = useSidecar();
+
+  let { containerProps, primaryProps, secondaryProps, splitterProps } = useSplitter({
+    direction: 'row',
+    initialSize: 0.6,
+    dragPosition: 'end',
+    allowOverflow: true,
+  });
+
+  const styles = useStyles2(getStyles);
+  const memoryLocationService = new HistoryWrapper(H.createMemoryHistory({ initialEntries: ['/'] }));
+
+  return (
+    <div {...(activePluginId ? containerProps : { className: 'grafana-app' })}>
+      <div {...(activePluginId ? primaryProps : { className: 'grafana-app' })}>
+        <RouterTree routes={props.routes} />
+      </div>
+      {/* Sidecar */}
+      {activePluginId && (
+        <>
+          <div {...splitterProps} />
+          <div {...secondaryProps}>
+            <Router history={memoryLocationService.getHistory()}>
+              <LocationServiceProvider service={memoryLocationService}>
+                <CompatRouter>
+                  <ModalsContextProvider>
+                    <GlobalStyles />
```

Comment:
won't this mean GlobalStyles is rendered twice?

Reply:
Ah yeah got it. I think you are right this should be shuffled around a because these contexts should be shared don't think there should be 2 of them and also as nav and toolbar are defined in AppChrome and they should be on top of both this probably makes sense to move a bit so will take a look at that and also at the router context cause yeah not sure if overwriting context will also change it upstream or not.